src/dsp_tools/utils/authentication_client_live.py

```diff
@@ -0,0 +1,47 @@
+from dataclasses import dataclass
+from typing import Any
+
+import requests
+from loguru import logger
+
+from dsp_tools.models.exceptions import BadCredentialsError
+from dsp_tools.models.exceptions import PermanentConnectionError
+from dsp_tools.models.exceptions import UserError
+from dsp_tools.utils.authentication_client import AuthenticationClient
+
+
+@dataclass
+class AuthenticationClientLive(AuthenticationClient):
+    """
+    Client that can authenticate with a DSP server and return a token.
+    """
+
+    server: str
+    email: str
+    password: str
+    _token: str | None = None
+
+    def get_token(self) -> str:
+        """
+        Returns a token. If no token is available, it will get one from the server.
+        """
+        if self._token:
+            return self._token
+        return self._get_token()
+
+    def _get_token(self) -> str:
+        url = f"{self.server}/v2/authentication"
+        payload = {"email": self.email, "password": self.password}
+        logger.info(f"Requesting token from url '{url}' for user '{self.email}'.")
```

Comment:
I see your point but I'm not really inclined to follow it. Part of why we wanted to separate authentication out of the general connection was that it made logging nasty, so I think it's the wrong default to "want to do it the same way" here.  
Additionally, the biggest reason why we always have to ask "what request was sent" is because there is such a wide variety of things that may be sent through the general purpose connection to the API. On the other hand, with this authentication client we know exactly what request has been sent, and all the moving parts (apart from the password which we wouldn't log anyways) are contained in the log I added; and for the response, if we get an OK, we have a pretty clear idea what happened, and if not, an error is propagated outward. So I do in fact think we have all the necessary information, as it stands now.  
So I'm not too happy to invest work to generate code that will need to be maintained, without it adding any extra value.

However, I'm absolutely willing to change the log message, if that helps you navigate the logs. What I could see right away, to align it with the existing:
- change log level to `debug`
- Add prefix `REQUEST: `
- Add a log line in the OK case, with the prefix `RESPONSE: `

Would that address your concerns?

Reply:
> So I do in fact think we have all the necessary information, as it stands now.
You're right.

> Would that address your concerns?
Yes :-)